### PR TITLE
align validator fetch to latest version of API endpoint

### PR DESCRIPTION
**Pull request by fabrice-adamik**
Summary

**This PR updates the logic for fetching validators and transaction history, aligning with the latest API endpoint specifications and improving the handling of paginated responses.**

Key Changes

1. Validator Fetching Logic:

- 	Aligned the validator fetch logic to the latest API endpoint version.
- 	Removed the limit parameter from the validators API, as it is no longer required.
- 	Introduced pagination support to handle large sets of validators efficiently.
- 	Filtered out non-staking chains before making API calls to optimize the validator list. 
- 	Added a helper function `isStakingSupported` to identify staking-supported chains. Also updated supported-chains labels to leverage this function.

2. Pagination Enhancements:

- Implemented pagination support for both validator fetching and transaction history.
- Added a “Load More” button for loading additional data when more pages are available.
- Enhanced the UI to remember the scroll position when fetching new transactions, improving the user experience.

3. UI Improvements:
- Updated the staking positions display to show accurate on the staking page
- Refined the transaction history display to handle paginated data and streamline the data presentation.

### CODE REVIEW

File: src/hooks/useValidators.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
 type GetValidatorParams = {
   chainId: string;
 };
 
 export const useValidators = ({ chainId }: GetValidatorParams) => {
   return useQuery({
-    queryKey: ["validators", chainId],
-    queryFn: async () => getValidators(chainId),
+    queryKey: ["validators", chainId, "all"],
```
Comment (hakim-adamik) [inline src/hooks/useValidators.tsx:10]: I don't think `"all"` is necessary here? Probably doesn't hurt either...

File: src/hooks/useValidatorsBatch.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+import { useQueries } from "@tanstack/react-query";
+import { getAllValidators } from "~/api/adamik/validators";
 
 export const useValidatorsBatch = (chainIds: string[]) => {
   return useQueries({
     queries: chainIds.map((chainId) => {
       return {
-        queryKey: ["validators", chainId],
-        queryFn: async () => getValidators(chainId, 0, 500),
+        queryKey: ["validators", chainId, "all"],
```
Comment (hakim-adamik) [inline src/hooks/useValidatorsBatch.tsx:8]: Same here

File: src/app/stake/page.tsx [hunk 3]
```diff
@@ -74,8 +75,13 @@ export default function Stake() {
     "symbols"
   );
 
+  const stakingSupportedChainIds = addressesChainIds.filter((chainId) => {
+    const chain = chainsDetails?.find((c) => c.id === chainId);
+    return chain && isStakingSupported(chain);
+  });
```
Comment (hakim-adamik) [inline src/app/stake/page.tsx:81]: A bit simpler + [useMemo](https://react.dev/reference/react/useMemo):
```suggestion
  const stakingSupportedChainIds = useMemo(() => chainsDetails?.find((chain) => isStakingSupported(chain)).map((chain) => chain.id);
  }),
  [chainDetails]);
```

File: src/app/stake/StakingPositionSelector.tsx [hunk 4]
```diff
@@ -188,7 +194,6 @@ const StakingPositionView = ({
   mode: TransactionMode;
 }) => {
   const validator = useMemo(() => {
-    // FIXME Hack for Cosmos, all validatorAddresses should be handled not just the 1st one
```
Comment (hakim-adamik) [inline src/app/stake/StakingPositionSelector.tsx:191]: :eyes: :wink: :pray:

File: src/app/history/page.tsx [hunk 5]
```diff
@@ -147,19 +153,57 @@ function TransactionHistoryContent() {
...
-      setTransactionHistory(null);
     } finally {
       setIsFetchingHistory(false);
     }
   };
 
-  // Add state for formatted amounts
+  const transactionListRef = useRef<HTMLDivElement>(null);
+
+  const handleLoadMore = async () => {
```
Comment (hakim-adamik) [inline src/app/history/page.tsx:175]: You should use [useCallback](https://react.dev/reference/react/useCallback) for this function

File: src/api/adamik/history.ts [hunk 6]
```diff
@@ -7,34 +7,30 @@ interface AccountHistoryResponse {
   chainId: string;
   accountId: string;
   transactions: FinalizedTransaction[];
+  pagination?: {
+    nextPage: string | null;
+  };
 }
 
 export const getAccountHistory = async (
```
Comment (hakim-adamik) [inline src/api/adamik/history.ts:15]: This function correctly sets `nextPage` if provided, however it doesn't read it from the response and does not loop while there is a `nextPage` token.

You just need the same as for validators: the `do {} while` loop to iterate as long as a `nextPage` is provided.

Comment (fabrice-adamik) [reply src/api/adamik/history.ts:15]: It's kind of on purpose... 

1. I only fetch the first page
2. fetch the other page only when the user click on Load More

This avoids unnecessary calls.

**Review by hakim-adamik — APPROVED** [review]

**Pull request merged**